### PR TITLE
fix(Checkbox): Remove empty label element

### LINKED ISSUE #1740: Fix empty `label` in `Checkbox`

### PR DESCRIPTION
**Pull request by poi33**
New styling that changed the label::before into input::before
Fixes #1740

### CODE REVIEW

**Comment by poi33** [comment]
This is an attempt to fix: https://github.com/digdir/designsystemet/issues/1735. 
Only the checkbox side.
Should be possible to apply the same changes to the radio input.

**Comment by Barsnes** [comment]
Hi, thanks for the PR!
I'll let @mimarz look at this, but be is currently on vacation until next week 🐣

**Review by mimarz — CHANGES_REQUESTED** [review]
Awesome!

First of thanks for wanting to contribute! 🤝 
 
There is a lot of good improvements here ❤️ 

Here is some general feedback.

1. Is it possible to solve this just using `.input` instead of `.input::before`? This might make it easier to solve next bulletpoints.
1. Focus styling looks broken. This is probably due to browsers native outline styling for `input` elements which needs to be overriden now.
image
1. We aim for [Target Size AAA](https://www.w3.org/WAI/WCAG22/Understanding/target-size-enhanced.html) on our selection controls (radio, checkboxes, switches etc.) which is missing now. Maybe this can be solved using `:before` & `:after` to get that target size back (`2.75rem`)? 🤔 
image

1. We use[ Github keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) for linking our PRs. This needs to be used in the description of the PR for it to work. I made a separate task for just fixing this in Checkbox, so you can link it to #1740

File: packages/react/src/components/form/Checkbox/Checkbox.module.css [hunk 1]
```diff
@@ -5,106 +5,113 @@
   --fds-checkbox-border-color: var(--fds-semantic-border-input-default);
   --fds-checkbox-border__hover: 0 0 0 6px var(--fds-semantic-surface-info-subtle-hover);
 
-  position: relative;
+  display: grid;
+  grid-template-columns: var(--fds-checkbox-size) auto;
+  gap: 6px 6px;
```
[the remark this answers is not in the document]
Comment (poi33) [reply packages/react/src/components/form/Checkbox/Checkbox.module.css]: Should be fixed in e12068b

**Comment by poi33** [comment]
@mimarz added new commits

1. Changed to use input instead of input::before for the styling
2. Removed outline from input element
3. The input::before now is set to a 44px (2.75rem) size. (Had some trouble with the hover reflecting this)
4. Updated to use github keywords. (also added the git issue number to the commit header)

File: packages/react/src/components/form/Checkbox/Checkbox.module.css [hunk 2]
```diff
@@ -18,93 +56,71 @@
...
 
-.input:focus-visible + .label::before {
+.readonly > .label,
+.readonly > .input,
+.readonly > .input::before {
+  cursor: default;
+}
+
+.input:focus-visible {
   outline: var(--fds-checkbox-focus-border-width) solid var(--fds-semantic-border-focus-outline);
```
Comment (mimarz) [inline packages/react/src/components/form/Checkbox/Checkbox.module.css:84]: You probably need to also add `outline-offset: 0;` to get the correct focus style now with the styling changes to use `input`. 

In this PR:
image

Design/Current:
image

**Review by mimarz — CHANGES_REQUESTED** [review]
Some fine bit of work here! Found a small bug still with focus styling.

File: packages/react/src/components/form/Checkbox/Checkbox.module.css (hunk 2 above)
Comment (poi33) [reply packages/react/src/components/form/Checkbox/Checkbox.module.css:84]: Forgot to test the state cross browser. 😞 
I'll add the fix now

Comment (mimarz) [reply packages/react/src/components/form/Checkbox/Checkbox.module.css:84]: ah, yeah, its easy to forget test other browsers 😅

File: packages/react/src/components/form/Checkbox/Checkbox.module.css [hunk 3]
```diff
@@ -116,19 +133,19 @@
...
 
-  .container:not(.readonly, .disabled) > .input:hover:checked + .label::before {
+  .container:not(.readonly, .disabled) > .input:hover:checked {
     --fds-checkbox-border-color: var(--fds-semantic-border-input-hover);
 
     box-shadow: var(--fds-checkbox-border__hover), inset 0 0 0 2px var(--fds-checkbox-border-color);
   }
 
-  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible + .label::before {
+  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible::before {
```
Comment (mimarz) [inline packages/react/src/components/form/Checkbox/Checkbox.module.css:148]: This looks abit weird 😅 

image

Looks like missed removing a `::before` when changing to styling on `input` element.

I think this suggestion should fix it 🤔 
```suggestion
  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible {
```

**Review by mimarz — CHANGES_REQUESTED** [review]

**Review by mimarz — APPROVED** [review]
Jolly good work sir! 
Thank your for your contribution, we appreciate it! ❤️ 🎉  

welcome-to-the-society-of-uncompromising-men-cheers

**Pull request merged**